src/service/models.py

```diff
@@ -0,0 +1,149 @@
+"""
```

Comment:
structurally it looks good to me. But without tests, I didn't examine the logic for each op.

Reply:
Not quite sure what you mean - you can check the logic of the code without tests...